src/App.tsx

```diff
@@ -3,34 +3,39 @@ import { BrowserRouter } from 'react-router-dom';
 import { QueryParamProvider } from 'use-query-params';
 import { ReactRouter5Adapter } from 'use-query-params/adapters/react-router-5';
 import { AppFooter, AppHeader, DerivIframe } from '@/components';
-import { useRedirectToOauth } from '@/hooks';
+import { useRedirectToOauth, useTrackjs } from '@/hooks';
 import AppContent from '@/routes/AppContent';
 import { initializeI18n, TranslationProvider } from '@deriv-com/translations';
 import { Loader, useDevice } from '@deriv-com/ui';
+import { ErrorBoundary } from './components/ErrorBoundary';
 
 const { VITE_CROWDIN_BRANCH_NAME, VITE_PROJECT_NAME, VITE_TRANSLATIONS_CDN_URL } = import.meta.env;
 const i18nInstance = initializeI18n({
     cdnUrl: `${VITE_TRANSLATIONS_CDN_URL}/${VITE_PROJECT_NAME}/${VITE_CROWDIN_BRANCH_NAME}`,
 });
 
 const App = () => {
+    const { init: initTrackJS } = useTrackjs();
     const { isDesktop } = useDevice();
     const { redirectToOauth } = useRedirectToOauth();
 
     redirectToOauth();
+    initTrackJS();
 
     return (
         <BrowserRouter>
-            <QueryParamProvider adapter={ReactRouter5Adapter}>
-                <TranslationProvider defaultLang='EN' i18nInstance={i18nInstance}>
-                    <Suspense fallback={<Loader isFullScreen />}>
-                        <DerivIframe />
-                        <AppHeader />
-                        <AppContent />
-                        {isDesktop && <AppFooter />}
-                    </Suspense>
-                </TranslationProvider>
-            </QueryParamProvider>
+            <ErrorBoundary fallback={<div>fallback component</div>}>
```

Comment:
@farrah-deriv what should we show to the user as the fallback component?

Reply:
We'll create an error component for this. Can you please add a TODO so we don't miss it?

Reply:
done 🙏